packages/data-context/test/unit/sources/FileDataSource.spec.ts

```diff
@@ -5,127 +5,273 @@ import path from 'path'
+          // and ./scripts is explicitly ignored
+          expect(files).to.have.length(2)
+        })
 
-    it('always ignores files within node_modules', async () => {
-      const nodeModulesPath = path.join(projectPath, 'node_modules')
+        it('converts globs to POSIX paths on windows', async () => {
+          const windowsSeperator = '\\'
 
-      await fs.mkdir(nodeModulesPath)
-      await fs.writeFile(path.join(nodeModulesPath, 'module-script-1.js'), '')
-      await fs.writeFile(path.join(nodeModulesPath, 'module-script-2.js'), '')
+          sinon.stub(os, 'platform').returns('win32')
+          const toPosixStub = sinon.stub(fileUtil, 'toPosix').callsFake((path) => {
+            return toPosixStub.wrappedMethod(path, windowsSeperator)
+          })
 
-      const files = await fileDataSource.getFilesByGlob(
-        projectPath,
-        '**/*script-*.js',
-        { ignore: ['./scripts/**/*'] },
-      )
+          const files = await fileDataSource.getFilesByGlob(
+            projectPath,
+            `**${windowsSeperator}*script-*.js`,
+          )
 
-      // only scripts at root should be found, as node_modules is implicitly ignored
-      // and ./scripts is explicitly ignored
-      expect(files).to.have.length(2)
+          expect(files).to.have.length(5)
+        })
+
+        it('finds files using given globby options', async () => {
+          const files = await fileDataSource.getFilesByGlob(
+            projectPath,
+            'root-script-*.js',
+            { absolute: false },
+          )
+
+          expect(files).to.have.length(2)
+          expect(files[0]).to.eq('root-script-1.js')
+          expect(files[1]).to.eq('root-script-2.js')
+        })
+      })
     })
 
-    it('converts globs to POSIX paths on windows', async () => {
-      const windowsSeperator = '\\'
+    describe('unit', () => {
+      const ctx = createTestDataContext('open')
+
+      ctx.coreData.currentTestingType = 'e2e'
+
+      const fileDataSource = new FileDataSource(ctx)
+      const mockMatches = ['/mock/matches']
+      const defaultGlobbyOptions = {
+        onlyFiles: true,
+        absolute: true,
+        ignore: ['**/node_modules/**'],
+      }
 
-      sinon.stub(os, 'platform').returns('win32')
-      const toPosixStub = sinon.stub(fileUtil, 'toPosix').callsFake((path) => {
-        return toPosixStub.wrappedMethod(path, windowsSeperator)
+      let matchGlobsStub: sinon.SinonStub
+
+      beforeEach(() => {
+        matchGlobsStub = sinon.stub(FileDataSourceModule, 'matchGlobs').resolves(mockMatches)
+      })
+
+      afterEach(() => {
+        sinon.restore()
+      })
+
+      it('matches absolute patterns when working directory is root', async () => {
+        const files = await fileDataSource.getFilesByGlob(
+          '/',
+          '/cypress/e2e/**.cy.js',
+        )
+
+        expect(files).to.eq(mockMatches)
+        expect(matchGlobsStub).to.have.been.calledWith(
+          ['cypress/e2e/**.cy.js'],
```

Comment:
In 10.6.0, this test case would produce `.cypress/e2e/**.cy.js`, which is not a valid pattern.